src/ctx.ts

```diff
@@ -62,6 +62,11 @@ export class Ctx {
       initializationOptions: { clangdFileStatus: true },
       disableDiagnostics: this.config.disableDiagnostics,
       outputChannel,
+      middleware: {
+        provideOnTypeFormattingEdits: (document, position, _ch, options, token, next) => {
```

Comment:
maybe `if (ch == '\n') ch = '';` and then `next(..., ch, ...)`?
Just to make the change as narrow as possible.